app/models/manager_refresh/save_collection/helper.rb

```diff
@@ -50,9 +50,29 @@ def create_or_update_inventory!(inventory_collection, record_index, association)
       unique_index_keys = inventory_collection.manager_ref_to_cols
 
       association.find_each do |record|
-        # TODO(lsmola) the old code was able to deal with duplicate records, should we do that? The old data still can
-        # have duplicate methods, so we should clean them up. It will slow up the indexing though.
-        record_index[inventory_collection.object_index_with_keys(unique_index_keys, record)] = record
+        index = inventory_collection.object_index_with_keys(unique_index_keys, record)
+        if record_index[index]
+          # We have a duplicate in the DB, destroy it
+          record_for_destruction = nil
+
+          if record.respond_to?(:service)
```

Comment:
@Fryguy so my main issue was with orchestration stacks, where we create them as part of the service ordering. But they are also created by refresh. So in case that these will happen at once, I want to keep the one related to the service. Otherwise the relation to the service would be lost forever.

This is the thing that would be fixed with a unique index. :-)

The issue with VmWare vms is another one, but we do not have that elsewhere.